src/Rope.elm

```diff
@@ -1,41 +1,80 @@
-module Rope exposing (Rope(..), empty, flatFromList, one, toList)
+module Rope exposing (Rope, RopeLikelyFilled(..), empty, flatFromList, one, prependTo, prependToLikelyFilled, toList)
 
 {-| inspired by [miniBill/elm-rope](https://dark.elm.dmy.fr/packages/miniBill/elm-rope/latest/)
 -}
 
 
-type Rope a
+type alias Rope a =
+    Maybe (RopeLikelyFilled a)
+
+
+type RopeLikelyFilled a
     = Leaf (List a)
-    | Branch (List (Rope a))
+    | Branch (List (RopeLikelyFilled a))
 
 
 empty : Rope a
 empty =
-    Leaf []
+    Nothing
 
 
-one : a -> Rope a
+one : a -> RopeLikelyFilled a
 one onlyElement =
     Leaf [ onlyElement ]
 
 
+prependToLikelyFilled : Rope a -> RopeLikelyFilled a -> Rope a
+prependToLikelyFilled right leftLikelyFilled =
+    case right of
+        Nothing ->
+            Just leftLikelyFilled
+
+        Just rightLikelyFilled ->
+            Just (Branch [ leftLikelyFilled, rightLikelyFilled ])
```

Comment:
Maybe we could skip the `Maybe` and pattern match on the contents of `Leaf` or `Branch`?
```elm
prependToLikelyFilled : Rope a -> RopeLikelyFilled a -> Rope a
prependToLikelyFilled right leftLikelyFilled =
    case right of
        Leaf [] ->
            Just leftLikelyFilled

        Branch [] ->
            Just leftLikelyFilled

        _ ->
            Just (Branch [ leftLikelyFilled, right ])
```
I think that could be faster, but I'm unsure.

Reply:
Hmm. Runs slower.

I've also tried:
- Introduce a `One a` variant → slower
- Introduce a `Branch2 filled filled` → not decisive

now testing the `Empty` variant thing

Reply:
completely switching out `Branch (List filled)` for `Branch2 filled filled` and `Leaf (List a)` for `Leaf a` seemed to have done the trick, since a branch in the previous version pretty much never had more than 2 filleds per branch.

A nice extra is that it's now not a "likely filled" but a typesafe "filled".